Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/SyncKronosToShiftsController.cs

```diff
@@ -104,51 +104,122 @@ public ActionResult SyncKronosToShifts(string isRequestFromLogicApp)
-                await this.timeOffRequestsController.ProcessTimeOffRequetsAsync(isRequestFromLogicApp).ConfigureAwait(false);
-
-                this.telemetryClient.TrackTrace($"{Resource.SyncTimeOffRequestsFromShiftsToKronos} completed from {Resource.ProcessKronosToShiftsShiftsAsync} ");
+            // Sync TimeOff and TimeOffRequest only if Paycodes in kronos synced successfully.
+             if (isMapPayCodeTimeOffReasonsSuccessful)
+            {
+                try
+                {
+                    // Sync timeoff from kronos to shifts.
+                    await this.timeOffController.ProcessTimeOffsAsync(isRequestFromLogicApp).ConfigureAwait(false);
+
+                    this.telemetryClient.TrackTrace($"{Resource.ProcessTimeOffsAsync} completed from {Resource.ProcessKronosToShiftsShiftsAsync} ");
+                }
+#pragma warning disable CA1031 // Do not catch general exception types
+                catch (Exception ex)
+#pragma warning restore CA1031 // Do not catch general exception types
+                {
+                    this.telemetryClient.TrackException(ex);
+                }
+
+                try
+                {
+                    // Sync timeoff from kronos to shifts.
+                    await this.timeOffRequestsController.ProcessTimeOffRequestsAsync(isRequestFromLogicApp).ConfigureAwait(false);
+
+                    this.telemetryClient.TrackTrace($"{Resource.SyncTimeOffRequestsFromShiftsToKronos} completed from {Resource.ProcessKronosToShiftsShiftsAsync} ");
+                }
+#pragma warning disable CA1031 // Do not catch general exception types
+                catch (Exception ex)
+#pragma warning restore CA1031 // Do not catch general exception types
+                {
+                    this.telemetryClient.TrackException(ex);
+                }
+            }
+            else
+            {
+                this.telemetryClient.TrackTrace($"{Resource.MapPayCodeTimeOffReasonsAsync} status:  " + isMapPayCodeTimeOffReasonsSuccessful);
+            }
 
+             // sync Shifts from Kronos to Shifts only if OpenShiftRequest and SwapShiftRequest sync is successful.
+             if (isSwapShiftRequestSyncSuccessful && isOpenShiftRequestSyncSuccessful)
+            {
                 // Sync shifts from kronos to shifts.
                 await this.shiftController.ProcessShiftsAsync(isRequestFromLogicApp).ConfigureAwait(false);
 
                 this.telemetryClient.TrackTrace($"{Resource.ProcessShiftsAsync} completed from {Resource.ProcessKronosToShiftsShiftsAsync} ");
-
-                this.telemetryClient.TrackTrace($"{Resource.ProcessKronosToShiftsShiftsAsync} completed at: {DateTime.UtcNow.ToString("O", CultureInfo.InvariantCulture)}" + " for isRequestFromLogicApp: " + isRequestFromLogicApp);
             }
-            catch (Exception)
+
+             // Do not sync shifts from kronos to shifts. Log the status of OpenShiftRequest and SwapShiftRequest sync.
```

Comment:
Capitalization is all over the place. Pay attention to that.
"Kronos". "Shifts" for the product. "shift" for the abstract concept of a shift.

Reply:
@aosolis - ack'd - I have made sure to clarify all capitalization and comment content as well